posthog/migrations/0676_add_question_ids_to_surveys.py

```diff
@@ -0,0 +1,45 @@
+# Generated by Django 4.2.18 on 2025-02-25 19:29
+
+import uuid
+from django.db import migrations
+
+
+def add_question_ids_to_surveys(apps, schema_editor):
+    """
+    Add a unique ID to each question in existing surveys.
+    This ensures all questions have an ID for tracking responses.
+    """
+    Survey = apps.get_model("posthog", "Survey")
+
+    # Get all surveys
+    surveys = Survey.objects.all()
+
+    for survey in surveys:
+        # Skip surveys with no questions
+        if not survey.questions:
+            continue
+
+        # Add IDs to questions that don't have them
+        modified = False
+        for question in survey.questions:
+            if not question.get("id"):
+                question["id"] = str(uuid.uuid4())
+                modified = True
+
+        # Save the survey if any questions were modified
+        if modified:
+            survey.save(update_fields=["questions"])
```

Comment:
will this method also call `pre_save_survey`? prob not but better to be sure so we don't run the same code twice

Reply:
no, the `survey.save()` call in the migration won't trigger the pre_save_survey function. Django migrations use apps.get_model() to create a historical version of the model that's isolated from signal handlers. 

when we call save() on this historical model instance, it bypasses the signal system entirely. This is by design to prevent unexpected side effects during migrations.